nucliadb/src/nucliadb/search/api/v1/augment.py

```diff
@@ -82,78 +98,202 @@ async def _augment_endpoint(
             if paragraph is None:
                 continue
+
             augmented_paragraphs[paragraph_id.full()] = AugmentedParagraph(
                 text=paragraph.text,
                 # TODO: we need multiple calls to augmentor to fulfill this information
             )
 
         return AugmentResponse(
             resources=augmented_resources,
+            fields=augmented_fields,
             paragraphs=augmented_paragraphs,
         )
 
 
+def parse_deep_resource_augment(
+    item: AugmentResources,
+) -> tuple[list[ResourceProperties], list[ExtractedDataTypeName], list[ResourceProp]]:
+    show = []
+    show_extracted = False
+    extracted = []
+    select = []
+
+    _resource_prop_to_show = {
+        nucliadb_models.augment.ResourceProp.BASIC: ResourceProperties.BASIC,
+        nucliadb_models.augment.ResourceProp.ORIGIN: ResourceProperties.ORIGIN,
+        nucliadb_models.augment.ResourceProp.EXTRA: ResourceProperties.EXTRA,
+        nucliadb_models.augment.ResourceProp.RELATIONS: ResourceProperties.RELATIONS,
+        nucliadb_models.augment.ResourceProp.VALUES: ResourceProperties.VALUES,
+        nucliadb_models.augment.ResourceProp.ERRORS: ResourceProperties.ERRORS,
+        nucliadb_models.augment.ResourceProp.SECURITY: ResourceProperties.SECURITY,
+    }
+    _resource_prop_to_extracted = {
+        nucliadb_models.augment.ResourceProp.EXTRACTED_TEXT: ExtractedDataTypeName.TEXT,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_METADATA: ExtractedDataTypeName.METADATA,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_SHORTENED_METADATA: ExtractedDataTypeName.SHORTENED_METADATA,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_LARGE_METADATA: ExtractedDataTypeName.LARGE_METADATA,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_VECTOR: ExtractedDataTypeName.VECTOR,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_LINK: ExtractedDataTypeName.LINK,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_FILE: ExtractedDataTypeName.FILE,
+        nucliadb_models.augment.ResourceProp.EXTRACTED_QA: ExtractedDataTypeName.QA,
+    }
+    _resource_prop_to_prop = {
+        nucliadb_models.augment.ResourceProp.TITLE: ResourceTitle(),
+        nucliadb_models.augment.ResourceProp.SUMMARY: ResourceSummary(),
+        nucliadb_models.augment.ResourceProp.CLASSIFICATION_LABELS: ResourceClassificationLabels(),
+    }
+    for prop in item.select:
+        if prop in _resource_prop_to_show:
+            show.append(_resource_prop_to_show[prop])
+        elif prop in _resource_prop_to_extracted:
+            show_extracted = True
+            extracted.append(_resource_prop_to_extracted[prop])
+        elif prop in _resource_prop_to_prop:
+            select.append(_resource_prop_to_prop[prop])
+
+    if show_extracted:
+        show.append(ResourceProperties.EXTRACTED)
+
+    return (
+        show,
+        extracted,
+        select,  # type: ignore
```

Comment:
out of curiosity, what is the problem with the typing here and above?

Reply:
mypy confused with unions. I'll annotate a couple of things to fix it